src/chain_spec.rs

```diff
@@ -39,23 +39,102 @@ use alloc::string::String;
 mod light_sync_state;
 mod structs;
 
+use crate::chain::chain_information::{
+    BabeEpochInformation, ChainInformation, ChainInformationConsensus,
+};
+
+pub struct LightSyncState {
+    inner: light_sync_state::DecodedLightSyncState,
+}
+
+fn convert_epoch(epoch: &light_sync_state::BabeEpoch) -> BabeEpochInformation {
+    let epoch_authorities: Vec<_> = epoch
+        .authorities
+        .iter()
+        .map(|authority| crate::header::BabeAuthority {
+            public_key: authority.public_key,
+            weight: authority.weight,
+        })
+        .collect();
+
+    BabeEpochInformation {
+        epoch_index: epoch.epoch_index,
+        start_slot_number: Some(epoch.slot_number),
+        authorities: epoch_authorities,
+        randomness: epoch.randomness,
+        c: epoch.config.c,
+        allowed_slots: epoch.config.allowed_slots,
+    }
+}
+
+impl LightSyncState {
+    pub fn as_chain_information(&self) -> ChainInformation {
+        // Create a sorted list of all regular epochs that haven't been pruned from the sync state.
+        let mut epochs: Vec<_> = self
+            .inner
+            .babe_epoch_changes
+            .epochs
+            .iter()
+            .filter_map(|((_, block_num), epoch)| match epoch {
+                light_sync_state::PersistedEpoch::Regular(epoch) => Some((block_num, epoch)),
+                _ => None,
+            })
+            .collect();
+
+        epochs.sort_unstable_by_key(|(&block_num, _)| block_num);
+
+        // Get the latest two epochs.
+        let prev_epoch = &epochs[epochs.len() - 2].1;
+        let current_epoch = &epochs[epochs.len() - 1].1;
+
+        ChainInformation {
+            finalized_block_header: self.inner.finalized_block_header.clone(),
+            consensus: ChainInformationConsensus::Babe {
+                slots_per_epoch: std::num::NonZeroU64::new(prev_epoch.duration).unwrap(),
```

Comment:
```suggestion
                slots_per_epoch: NonZeroU64::new(prev_epoch.duration).unwrap(),
```

`use core::num::NonZeroU64` above. We don't use `std`.